### PR TITLE
#137005169 Embed social content

### PR DESCRIPTION
**Pull request by andela-jare**
#### What does this PR do?
This PR embeds disqus for products review.
#### Description of Task to be completed?
Embed reviews, tweets, and other social content.
#### How should this be manually tested?
On the website, click on a product to see its details and you will be able to comment on the product via disqus.
#### Any background context you want to provide?
None.
#### What are the relevant pivotal tracker stories?
#137005169.
#### Screenshots (if appropriate)
embed disqus
#### Questions:
None.

### CODE REVIEW

File: imports/plugins/included/product-detail-simple/client/templates/productDetailSimple.js [hunk 1]
```diff
@@ -9,3 +10,12 @@ Template.productDetailSimple.helpers({
     return ProductDetailContainer;
   }
 });
+Template.disqus.helpers({
+  getDisqus() {
+    const d = document;
```
Comment (andela-aolaniran) [inline imports/plugins/included/product-detail-simple/client/templates/productDetailSimple.js]: How about a descriptive variable name ?

File: imports/plugins/included/product-detail-simple/client/templates/productDetailSimple.js [hunk 2]
```diff
@@ -9,3 +10,12 @@ Template.productDetailSimple.helpers({
...
+    const s = d.createElement("script");
+    s.src = "//anbu-rc.disqus.com/embed.js";
```
Comment (andela-aolaniran) [inline imports/plugins/included/product-detail-simple/client/templates/productDetailSimple.js]: How about a descriptive variable name ?

File: server/security/policy.js [hunk 3]
```diff
@@ -22,13 +22,25 @@ if (process.env.NODE_ENV === "development") {
...
 
+BrowserPolicy.content.allowOriginForAll("*.disqus.com");
+BrowserPolicy.content.allowOriginForAll("https://disqus.com");
+BrowserPolicy.content.allowOriginForAll("http://disqus.com");
+BrowserPolicy.content.allowOriginForAll("https://a.disquscdn.com");
+BrowserPolicy.content.allowEval("*.disqus.com");
+
 BrowserPolicy.content.allowImageOrigin("fbcdn-profile-a.akamaihd.net");
 BrowserPolicy.content.allowImageOrigin("secure.gravatar.com");
 BrowserPolicy.content.allowImageOrigin("i0.wp.com");
```
Comment (andela-aolaniran) [inline server/security/policy.js:50]: Good work.

Comment (vonvick) [reply server/security/policy.js:50]: LGTM

Comment (vonvick) [reply server/security/policy.js:50]: Did not see other social media like facebook, twitter

File: imports/plugins/included/product-variant/client/templates/products/products.js [hunk 4]
```diff
@@ -172,3 +173,37 @@ Template.products.events({
...
+  this.state.setDefault({
+    timeline: {}
+  });
+
+  this.autorun(() => {
+    this.subscribe("Packages");
+    const socialConfig = Packages.findOne({
+      name: "reaction-social"
+    });
+    this.state.set("timeline", socialConfig.settings.public.apps);
```
Comment (andela-aolaniran) [inline imports/plugins/included/product-variant/client/templates/products/products.js:188]: What happens if socialConfig is 'undefined', you might want to put that into consideration.

File: imports/plugins/included/product-variant/client/templates/products/products.html [hunk 5]
```diff
@@ -34,3 +35,19 @@
...
+<template name="socialTimeline">
+  <div class="col-xs-12" style="z-index:10; padding:20px 60px;">
+    {{#if facebook}}
+    <div class="col-sm-6">
+      <iframe src={{facebook}} style="border:none;overflow:auto;width:100%;height:400px" allowTransparency="true"></iframe>
+    </div>
+    {{/if}}
+    {{#if twitter}}
+    <div class="col-sm-6" style="height: 400px; overflow-y: scroll;">
+      <a class="twitter-timeline" href="{{twitter}}">Tweets by {{twitter}}</a>
```
Comment (andela-aolaniran) [inline imports/plugins/included/product-variant/client/templates/products/products.html:48]: Is it possible not to use Inline styles ?

File: imports/plugins/included/product-variant/client/templates/products/products.js (hunk 4 above)
Comment (andela-jare) [reply imports/plugins/included/product-variant/client/templates/products/products.js:188]: Good point but socialConfig can't evaluate to undefined.

File: imports/plugins/included/product-variant/client/templates/products/products.js [hunk 6]
```diff
@@ -172,3 +172,37 @@ Template.products.events({
     loadMoreProducts();
   }
 });
+
+Template.socialTimeline.onCreated(function () {
+  this.state = new ReactiveDict();
```
[the remark this answers is not in the document]
Comment (andela-jare) [reply imports/plugins/included/product-variant/client/templates/products/products.js:178]: Fixed

File: imports/plugins/included/product-variant/client/templates/products/products.html (hunk 5 above)
Comment (andela-jare) [reply imports/plugins/included/product-variant/client/templates/products/products.html:48]: These elements are unique. Hence the use of inline styles.

Comment (andela-aolaniran) [reply imports/plugins/included/product-variant/client/templates/products/products.html:48]: Okay.

File: imports/plugins/included/product-variant/client/templates/products/products.js (hunk 4 above)
Comment (andela-aolaniran) [reply imports/plugins/included/product-variant/client/templates/products/products.js:188]: Ok.

**Review by vonvick — APPROVED** [review]
LGTM

**Review by andela-aolaniran — APPROVED** [review]
Looks good.

**Pull request merged**